File: build_artifact.py
#!/usr/bin/env python3
"""Baut aus web/artifact_template.html + data/latest.json die fertige
Uebersichtsseite telegram-monitor-uebersicht.html (eine Datei, offline nutzbar).

  python cli.py scan --json > /tmp/scan.json     # optional: frische Daten
  python build_artifact.py
"""
from __future__ import annotations

import json
import os
import sys

ROOT = os.path.dirname(os.path.abspath(__file__))
TEMPLATE = os.path.join(ROOT, "web", "artifact_template.html")
DATA = os.path.join(ROOT, "data", "latest.json")
OUT = os.path.join(os.path.dirname(ROOT), "telegram-monitor-uebersicht.html")


def build(data_path: str = DATA, out_path: str = OUT) -> str:
    with open(TEMPLATE, "r", encoding="utf-8") as fh:
        tpl = fh.read()
    with open(data_path, "r", encoding="utf-8") as fh:
        data = json.load(fh)
    payload = json.dumps(data, ensure_ascii=False).replace("</", "<\\/")
    html = tpl.replace("/*__DATA__*/{}", payload)
    with open(out_path, "w", encoding="utf-8") as fh:
        fh.write(html)
    return out_path


if __name__ == "__main__":
    path = build(sys.argv[1] if len(sys.argv) > 1 else DATA)
    print(f"geschrieben: {path} ({os.path.getsize(path)} Bytes)")

File: javascript/build_artifact.js
#!/usr/bin/env node
// build_artifact.py — portiert nach javascript
// Quelle: python, Projects@Telegram-Monitor:build_artifact.py
// Erzeugt: 2026-08-08 durch ABSTRACTIONS_MANAGER.py

/**
 * Baut aus web/artifact_template.html + data/latest.json die fertige
 * Uebersichtsseite telegram-monitor-uebersicht.html (eine Datei, offline nutzbar).
 *
 *   python cli.py scan --json > /tmp/scan.json     // optional: frische Daten
 *   node build_artifact.js
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const ROOT = __dirname;
const TEMPLATE = path.join(ROOT, "web", "artifact_template.html");
const DATA = path.join(ROOT, "data", "latest.json");
const OUT = path.join(path.dirname(ROOT), "telegram-monitor-uebersicht.html");

function build(dataPath = DATA, outPath = OUT) {
    const tpl = fs.readFileSync(TEMPLATE, 'utf-8');
    const data = JSON.parse(fs.readFileSync(dataPath, 'utf-8'));
    const payload = JSON.stringify(data, null, 0).replace(/<\/([^a-z])/g, '<\\/$1');
    const html = tpl.replace("/*__DATA__*/{}", payload);
    fs.writeFileSync(outPath, html, 'utf-8');
    return outPath;
}

if (import.meta.url === `file://${process.argv[1]}`) {
    const path = build(process.argv[2]);
    console.log(`geschrieben: ${path} (${fs.statSync(path).size} Bytes)`);
}
